distributed/sharded_tensor/tensor_parallel.py

```diff
@@ -0,0 +1,113 @@
+import os
+import tempfile
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+import torch.nn as nn
+	
+from torch.distributed._shard import shard_module
+from torch.distributed._shard.sharded_optim import (
+    ShardedOptimizer,
+    named_params_with_sharded_tensor,
+)
+from torch.distributed._shard.sharding_plan import ShardingPlan
+from torch.distributed._shard.sharding_spec import ChunkShardingSpec
+
+"""
+This is script to test Tensor Parallel(TP) on a toy model in a 
+Megetron-LM SPMD style. We show a E2E working flow from forward,
+backward and optimization.
```

Comment:
Add some more details here mentioning that we shard first linear layer col-wise, second one row-wise and avoid comms in between the two.